app/actions/handlers.py

```diff
@@ -1,28 +1,194 @@
+            integration=integration,
+            config=client.get_auth_config(integration)
+        )
+    except httpx.HTTPStatusError as e:
+        if e.response.status_code == 401:
+            return {"valid_credentials": False}
+        else:
+            logger.error(f"Error getting token from Cellstop API. status code: {e.response.status_code}")
+            raise e
+    except httpx.HTTPError as e:
+        message = f"auth action returned error."
+        logger.exception(message, extra={
+            "integration_id": str(integration.id),
+            "attention_needed": True
+        })
+        raise e
+    else:
+        logger.info(f"Authenticated with success. token: {token}")
+        return {"valid_credentials": token is not None}
+
+
+async def action_fetch_samples(integration, action_config: PullObservationsConfig):
+    logger.info(f"Executing fetch_samples action with integration {integration} and action_config {action_config}...")
+    try:
+        vehicles = await client.get_vehicles_positions(
+            integration=integration,
+            config=client.get_fetch_samples_config(integration)
+        )
+    except httpx.HTTPStatusError as e:
+        if e.response.status_code == 401:
+            message = f"Invalid credentials. Please review the integration configuration. Integration id: {integration.id}"
+            logger.warning(message, extra={"integration_id": str(integration.id)})
+            return {
+                "observations_extracted": 0,
+                "observations": [],
+                "error": "Invalid credentials. Please review the authentication configuration."
+            }
+        else:
+            logger.error(f"Error getting token from Cellstop API. status code: {e.response.status_code}")
+            raise e
+    except httpx.HTTPError as e:
+        message = f"fetch_samples action returned error."
+        logger.exception(message, extra={
+            "integration_id": str(integration.id),
+            "attention_needed": True
+        })
+        raise e
+    else:
+        logger.info(f"Observations pulled with success.")
+        return {
+            "observations_extracted": len(vehicles),
+            "observations": [json.loads(vehicle.json()) for vehicle in vehicles]
+        }
+
+
+@activity_logger()
+async def action_pull_observations(integration, action_config: PullObservationsConfig):
+    logger.info(f"Executing pull_observations action with integration {integration} and action_config {action_config}...")
+    try:
+        async for attempt in stamina.retry_context(
+                on=httpx.HTTPError,
+                attempts=3,
+                wait_initial=datetime.timedelta(seconds=10),
+                wait_max=datetime.timedelta(seconds=10),
+        ):
+            with attempt:
+                vehicles = await client.get_vehicles_positions(
+                    integration=integration,
+                    config=client.get_pull_config(integration)
```

Comment:
In the latest version of the template now you receive the parsed config in `action_config`, so you don't need to do `client.get_pull_config() `now.
```suggestion
                    config=action_config
```